lib/lcset.gi

```diff
@@ -0,0 +1,139 @@
+##############################################################################
+##
+#W  lcset.gi                 GAP4 package `Utils'                Chris Wensley
+##
+#Y  Copyright (C) 2015-2022, The GAP Group 
+
+#############################################################################
+##
+#O  LeftCoset( <g>, <U> ) . . . . . . . . forms a left coset of U by g \in G 
+##
+InstallMethod( LeftCoset, "generic method for left cosets", true, 
+    [ IsObject, IsGroup ], 0,
+function( g, U )
+    local lc;
+
+    lc := rec();
+    ObjectifyWithAttributes( lc, IsLeftCosetType,
+        Representative, g,
+        ActingDomain, U, 
+        IsLeftCoset, true );
+    return lc;
```

Comment:
To match `RightCoset` a little bit more, you could also insert something like this:
```suggestion
    if HasSize(U) then SetSize(lc, Size(U)); fi;
    return lc;
```

Reply:
Done.